### PR TITLE
Fix disappearing icons when browser reports a non-integer pixel ratio

### LINKED ISSUE #1029: Sprites on 1.5x
I have two custom sprite sheets, on for @1x and one for @2x, but my icons get cut off on displays with a pixel density of 1.5.

I tried making sprites@1.5x.json, but it didn't do anything different.

I did notice that some of the default Mapbox sprite sheets don't have this problem.  What am I doing wrong?

capture

### PR DESCRIPTION
**Pull request by lucaswoj**
resolves #1029 #1475 #1476

replaces #1481 

cc @jfirebaugh @fnicollet @kkaefer

### CODE REVIEW

**Comment by fnicollet** [comment]
Following up on #1481 and #1476, I tried your PR and while it works, the rendering is not great.
Here is the version with the ternary:
image


and my version which rounds up the pixel afterwards:
image

Both are at 82%

Here is the diff for the highway shields:
image

rounding up:
image

So while it works, it doesn't give the best results for POI icons, which become hardly readable.

**Comment by lucaswoj** [comment]
GL JS requires that all sprites are rendered at either 1x or 2x resolution. 

If you look closely at your first set of diffs (Jardin Commandant Pierre Aubertin) you'll notice that the "rounded up" sprite is incorrectly sized relative to the text. 

I think the "not great" rendering you're seeing is the result of the naive image scaling algorithm we are using (handily annotated with a `TODO` :grimacing:) https://github.com/mapbox/mapbox-gl-js/blob/master/js/symbol/sprite_atlas.js#L65-L66.

**Comment by fnicollet** [comment]
Ahah yes I saw that TODO in the code before, it is probably the cause of this aliasing.
Anyway, your fix should do for now. My case is a bit special (Windows 10 app have a 82% scaling by default for some reason) but this is not an issue for most people, there are some other issues to focus on :)

File: js/symbol/sprite_atlas.js [hunk 1]
```diff
@@ -38,6 +38,8 @@ SpriteAtlas.prototype = {
 };
 
 SpriteAtlas.prototype.resize = function(newRatio) {
+    newRatio = newRatio > 1 ? 2 : 1;
```
Comment (1ec5) [inline js/symbol/sprite_atlas.js]: Is it possible for the ratio to be less than 1 if, say, you zoom out below 100% (with the zoom only text setting disabled)? Or is the ratio independent of the browser's zoom level?

Comment (lucaswoj) [inline js/symbol/sprite_atlas.js]: There's no standardization. Some browsers will change the ratio with the zoom level, some will not. The MS Surface Pro reports a default ratio of 0.87.

**Pull request merged**